src/classes/OPP_PrimaryContactRoleMerge.cls

```diff
@@ -0,0 +1,256 @@
+        }
+    }
+
+
+    /*******************************************************************************************************
+    * @description It encapsulates an Opportunity and its related duplicate Primary OCRs.
+    ********************************************************************************************************/
+    public class OpportunityPrimaryOCRs {
+        private Opportunity record;
+
+        /*********************************************************************************************************
+        * @description The Constructor of OpportunityPrimaryOCRs.
+        * @param Opportunity Opportunity with the related primary OCRs.
+        */
+        public OpportunityPrimaryOCRs(Opportunity currentOpportunity) {
+            record = currentOpportunity;
+        }
+
+        /*******************************************************************************************************
+        * @description Checks if the Opportunity has an OCR Role equals to the Role in the parameter.
+        * @param String OCR Role.
+        * @return Boolean.
+        ********************************************************************************************************/
+        public Boolean containsRole(String ocrRole) {
+            for (OpportunityContactRole ocr : record.OpportunityContactRoles) {
+                if (ocr.Role == ocrRole) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /*******************************************************************************************************
+        * @description Gets the OCRs with role equals blank.
+        * @return List<OpportunityContactRole> The OCRs with role role = blank.
+        ********************************************************************************************************/
+        public List<OpportunityContactRole> getBlankRole() {
+            List<OpportunityContactRole> result = new List<OpportunityContactRole>();
+            for (OpportunityContactRole ocr : record.OpportunityContactRoles) {
+                if (String.isBlank(ocr.Role)) {
+                    result.add(ocr);
+                }
+            }
+            return result;
+        }
+
+        /*******************************************************************************************************
+        * @description Gets the OCRs with Role equals to the Role in the parameter.
+        * @param String OCR Role.
+        * @return List<OpportunityContactRole> The OCRs with Role equals to the Role in the parameter.
+        ********************************************************************************************************/
+        public List<OpportunityContactRole> getRoleEquals(String ocrRole) {
+            List<OpportunityContactRole> result = new List<OpportunityContactRole>();
+            for (OpportunityContactRole ocr : record.OpportunityContactRoles) {
+                if (ocr.Role == ocrRole) {
```

Comment:
Case sensitivity could be an issue here.

Reply:
ocrRole.equalsIgnoreCase(ocr.Role)